specreduce/calibration_data.py

```diff
@@ -62,28 +52,81 @@
     "spechayescal"
 ]
 
+PYPEIT_CALIBRATION_LINELISTS = [
+    'Ne_IR_MOSFIRE',
+    'ArII',
+    'CdI',
+    'OH_MOSFIRE_H',
+    'OH_triplespec',
+    'Ar_IR_MOSFIRE',
+    'OH_GNIRS',
+    'ThAr_XSHOOTER_VIS',
+    'ThAr_MagE',
+    'HgI',
+    'NeI',
+    'XeI',
+    'OH_MODS',
+    'ZnI',
+    'OH_GMOS',
+    'CuI',
+    'ThAr_XSHOOTER_VIS_air',
+    'ThAr_XSHOOTER_UVB',
+    'OH_NIRES',
+    'HeI',
+    'FeI',
+    'OH_MOSFIRE_J',
+    'KrI',
+    'Cd_DeVeny1200',
+    'Ar_IR_GNIRS',
+    'OH_MOSFIRE_Y',
+    'ThAr',
+    'FeII',
+    'OH_XSHOOTER',
+    'OH_FIRE_Echelle',
+    'OH_MOSFIRE_K',
+    'OH_R24000',
+    'Hg_DeVeny1200',
+    'ArI'
+]
 
-def get_reference_file_path(path=None, cache=False, show_progress=False):
+
+def get_reference_file_path(
```

Comment:
Instead of rolling your own here, why not use the stuff in `astropy.utils.data` but use your own [conf.dataurl](https://github.com/astropy/astropy/blob/b40081ff722c5ca0a4ff1578abe4587e44c0eeba/astropy/utils/data.py#L99)?

https://github.com/astropy/astropy/blob/main/astropy/utils/data.py

Reply:
when this was originally implemented years ago, i followed examples in https://docs.astropy.org/en/stable/utils/data.html and just wrapped `astropy.utils.data.download_file()` with some error handling. it looks like those examples haven't changed much since. i'll look into refactoring this to use pure-`astropy` at a higher level. that's out of scope for this PR, though.

Reply:
i opened #169 to cover this. it's a little unclear to me how best to handle multiple data URLs with the `conf.dataurl` scheme. will play around more when i get the chance.